### PR TITLE
[B+C] Create InventoryPickupItemEvent. Adds BUKKIT-3798

### PR DESCRIPTION
**Pull request by DeltaWhy**
This event is called when a hopper or hopper minecart picks up a dropped item.

CraftBukkit: https://github.com/Bukkit/CraftBukkit/pull/1063
Leaky: https://bukkit.atlassian.net/browse/BUKKIT-3798

This depends on Bukkit#797: https://github.com/Bukkit/Bukkit/pull/797
and on CraftBukkit#1061: https://github.com/Bukkit/CraftBukkit/pull/1061

### CODE REVIEW

File: src/main/java/org/bukkit/event/inventory/InventoryPickupItemEvent.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+public class InventoryPickupItemEvent extends Event implements Cancellable {
+    private static final HandlerList handlers = new HandlerList();
+    private boolean cancelled;
+    private final Inventory inventory;
+    private final Item item;
+
+    public InventoryPickupItemEvent(final Inventory inventory, final Item item) {
+	super();
+	this.inventory = inventory;
+	this.item = item;
```
Comment (natemort) [inline src/main/java/org/bukkit/event/inventory/InventoryPickupItemEvent.java]: Formatting is off.

File: src/main/java/org/bukkit/event/inventory/InventoryPickupItemEvent.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	this.cancelled = cancel;
+    }
+
+    @Override
+    public HandlerList getHandlers() {
+	return handlers;
+    }
+
+    public static HandlerList getHandlerList() {
+	return handlers;
```
Comment (natemort) [inline src/main/java/org/bukkit/event/inventory/InventoryPickupItemEvent.java]: The method body for this method and the three above it all have incorrect indentation.

**Comment by DeltaWhy** [comment]
Fixed indentation. I will figure this out eventually.

**Comment by riking** [comment]
Should this extend EntityPickupEvent?

**Comment by amaranth** [comment]
Pulled, thanks.

**Pull request closed without merging**